test/README.md

```diff
@@ -17,4 +17,34 @@ t.Run("foo equals foo", func(t *testing.T) {
 t.Run("foo does not equal foo", func(t *testing.T) {
     test.NotEquals(t, "foo", "bar")
 })
-```
\ No newline at end of file
+```
+
+### Using the ErrorTypeComparer with `cmp`
+
+This comparer relies on the use of the excellent [go-cmp](https://github.com/google/go-cmp) library.
+
+As it's documentation states, it is intended to be a more powerful and safer alternative to reflect.DeepEqual for comparing whether two values are semantically equal.
+
+See [GoDoc documentation](https://godoc.org/github.com/google/go-cmp/cmp) for more information.
+
+```go
+t.Run("some error is not a test error", func(t *testing.T) {
+	// some fake errors, for the purpose of this example.
+	testErr := TestError{errors.New("oops")}
+	someErr := errors.New("ouch")
+
+	var e test.ErrorReporter
+	opts := cmp.Options{
+		test.ErrorTypeComparer,
+		cmp.Reporter(&e),
+	}
+	if !cmp.Equal(testErr, someErr, opts) {
+		fmt.Println(e.String())
+	}
```

Comment:
Honestly, this feels like quite a lot of boilerplate. I'm wondering if we should provide functions that gave us the error reporter and options as a tuple.

```go
r, opts := test.ReportErrors()
if !cmp.Equal(expected, err, opts) {
    fmt.Println(r.String())
}

```